plugins/modules/openssl_signature.py

```diff
@@ -0,0 +1,383 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Patrick Pichler <ppichler+ansible@mgit.at>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: openssl_signature
+short_description: Sign and verify data with openssl
+description:
+    - This module allows one to sign and verify data via a certificate and a private key
+    - The module can use the cryptography Python library, or the pyOpenSSL Python
+      library. By default, it tries to detect which one is available. This can be
+      overridden with the I(select_crypto_backend) option. Please note that the PyOpenSSL backend
+      was deprecated in Ansible 2.9 and will be removed in Ansible 2.13.
+requirements:
+    - Either cryptography >= 1.4 (some key types require newer versions)
+    - Or pyOpenSSL >= 0.11 (Ed25519 and Ed448 keys are not supported with this backend)
+author:
+    - Patrick Pichler (@aveexy)
+    - Markus Teufelberger (@MarkusTeufelberger)
+options:
+    action:
+        description: Action to be executed
+        type: str
+        required: true
+        choices: [ sign, verify ]
```

Comment:
Conceptually, I think it would be better to have two modules, one for signing, the other for verifying. I think the idea of having modules which do different actions depending on a module option is nowadays discouraged in Ansible.

How about having something like `openssl_signature_sign` and `openssl_signature_verify`?

Reply:
I'm not opposed, though the content to boilerplate/docs ratio there would be even higher.

Instead of `openssl_signature_verify` there could also be an `openssl_signature_info` module that tells you if a signature is valid/verified? That would follow the current pattern here maybe even more...

Reply:
Yep, it definitely would increase the content/boilerplate ratio.

`_info` is probably better. I first thought about that too, and was wondering whether it is a good fit and first decided against it, but now (after thinking some more about it) I guess it's better.

Reply:
Alright, I'll go the `openssl_signature`/`openssl_signature_info` route. Time to train those Ctrl + c / Ctrl + v skills! :-D